execution/gethexec/sync_monitor.go

```diff
@@ -70,41 +75,43 @@ func (s *SyncMonitor) SyncProgressMap() map[string]interface{} {
 }
 
 func (s *SyncMonitor) SafeBlockNumber(ctx context.Context) (uint64, error) {
```

Comment:
not necessarily for this PR - with push we should be able to remove SafeBlockNumber and FinalizedBlockNumber entirely.
These functions are only used by geth in arbitrum/apibackend, but with push we can just push them to blockchain and read from there, like the normal ethereum API.

Reply:
Done.